devito/symbolics/extended_sympy.py

```diff
@@ -202,6 +202,52 @@ def base(self):
     __reduce_ex__ = Pickable.__reduce_ex__
 
 
+class ExternalFunctionCall(sympy.Expr, Pickable):
```

Comment:
Why is this an Expr and not a sympy function?

Reply:
It can have an arbitrary number of arguments

Reply:
So does a sympy function. A sympy function is literraly `func.name(*args)` which you are rebuilding in some ways rather than using it

Reply:
You're right - a simple `sympy.Function` seems to work just fine. I was under the impression it didn't work for some reason :/